src/main/java/org/folio/circulation/resources/CirculationRulesResource.java

```diff
@@ -115,14 +144,84 @@ private void put(RoutingContext routingContext) {
     LoanCirculationRulesEngineResource.clearCache(new WebContext(routingContext).getTenantId());
     RequestCirculationRulesEngineResource.clearCache(new WebContext(routingContext).getTenantId());
 
-    loansRulesClient.put(rulesInput.copy()).thenAccept(result ->
-      result.applySideEffect(response -> {
+    clients.circulationRulesStorage().put(rulesInput.copy())
+      .thenAccept(res -> res.applySideEffect(response -> {
         if (response.getStatusCode() == 204) {
           SuccessResponse.noContent(routingContext.response());
         } else {
           ForwardResponse.forward(routingContext.response(), response);
         }
-    }, cause -> cause.writeTo(routingContext.response())));
+      }, cause -> cause.writeTo(routingContext.response())));
+  }
+
+  private void validatePolicy(Map<String, Set<String>> existingPoliciesIds, String policyType, List<CirculationRulesParser.PolicyContext> policies, Token token) {
+    if (!policyExists(existingPoliciesIds, policyType, policies)) {
+       throw new CirculationRulesException(
+         String.format("The policy %s does not exist", policyType),
+         token.getLine(), token.getCharPositionInLine());
+    }
+  }
+
+  private boolean policyExists(Map<String, Set<String>> existingPolicyIds,
+    String policyType, List<CirculationRulesParser.PolicyContext> policies) {
+
+    return MapUtils.isNotEmpty(existingPolicyIds) && isNotEmpty(existingPolicyIds.get(policyType))
+        && existingPolicyIds.get(policyType).contains(
+          policies.get(FIRST_ELEMENT_OF_LIST)
+            .getChild(POLICY_ID_POSITION_NUMBER).getText());
+  }
+
+  private CompletableFuture<Result<Map<String, Set<String>>>> getExistingIdsOfSpecificPolicy(Clients clients) {
```

Comment:
This appears to get a map of all policy IDs by type? Is that a reasonable description of the behaviour?

Reply:
yes, it is

Reply:
Maybe it could be renamed to something like `getPolicyIdsByType`?

Reply:
renamed